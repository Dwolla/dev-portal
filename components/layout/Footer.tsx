import styled from "@emotion/styled";
import Link from "next/link";
import map from "lodash.map";
import { HEADLINE_TEXT, PARAGRAPH_TEXT } from "../colors";
import { POPPINS, ROBOTO } from "../typography";
import { breakUp, breakDown } from "../breakpoints";
import logo from "../../assets/images/dwolla-developers-logo.svg";

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;

  @media (${breakUp("md")}) {
    padding: 20px;
  }

  @media (${breakUp("xxl")}) {
    max-width: 70%;
  }
`;

const LogoContainer = styled.div`
  width: 35%;
  padding: 20px;

  @media (${breakDown("sm")}) {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
  }
`;

const Logo = styled.img`
  width: 173px;
`;

const Copyright = styled.div`
  font-family: ${ROBOTO};
  font-size: 12px;
  line-height: 21px;
  color: ${PARAGRAPH_TEXT};
  margin-top: 14px;
`;

const LinkContainer = styled.div`
  width: 65%;
  display: flex;
  justify-content: flex-end;

  @media (${breakDown("md")}) {
    width: 100%;
    justify-content: space-between;
    flex-wrap: wrap;
    order: -1;
  }

  @media (${breakDown("sm")}) {
    order: unset;
  }
`;

const LinkGroup = styled.div`
  padding: 20px;

  @media (${breakUp("lg")}) {
    width: 200px;
  }
`;

const LinkGroupHeading = styled.h3`
  font-family: ${POPPINS};
  font-weight: 600;
  color: ${HEADLINE_TEXT};
  text-transform: uppercase;
  font-size: 11px;
`;

const LinkGroupList = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
`;

const LinkGroupItem = styled.li`
  margin: 0;
`;

const LinkGroupLink = styled.a`
  display: block;
  padding: 6px 0;
  font-family: ${ROBOTO};
  font-size: 14px;
  line-height: 20px;
  color: ${PARAGRAPH_TEXT};
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;

const LegalContainer = styled.div`
  width: 70%;
  padding: 20px;

  @media (${breakDown("sm")}) {
    width: 100%;
  }
`;

const LegalTitle = styled.strong`
  font-family: ${ROBOTO};
  font-weight: 600;
  color: ${HEADLINE_TEXT};
  font-size: 12px;
`;

const LegalDesc = styled.p`
  font-family: ${ROBOTO};
  font-size: 12px;
  line-height: 21px;
  color: ${PARAGRAPH_TEXT};
`;

export interface FooterLink {
  text: string;
  href: string;
}

interface FooterProps {
  links: Record<string, FooterLink[]>;
  legal: {
    title: string;
    veridianDescription: string;
    metaBankDescription: JSX.Element;
  };
}

export default function Footer(props: FooterProps) {
  return (
    <Container>
      <LogoContainer>
        <Link href="/">
          <a>
            <Logo src={logo} alt="" />
          </a>
        </Link>
        <Copyright>{new Date().getFullYear()} All Rights Reserved</Copyright>
      </LogoContainer>

      <LinkContainer>
        {map(props.links, (v: FooterLink[], k: string) => (
          <LinkGroup key={k}>
            <LinkGroupHeading>{k}</LinkGroupHeading>

            <LinkGroupList>
              {v.map((i) => (
                <LinkGroupItem key={i.href}>
                  <LinkGroupLink href={i.href}>{i.text}</LinkGroupLink>
                </LinkGroupItem>
              ))}
            </LinkGroupList>
          </LinkGroup>
        ))}
      </LinkContainer>

      <LegalContainer>
        <LegalTitle>{props.legal.title}</LegalTitle>
        <LegalDesc>{props.legal.veridianDescription}</LegalDesc>
        <LegalDesc>{props.legal.metaBankDescription}</LegalDesc>
      </LegalContainer>
    </Container>
  );
}
