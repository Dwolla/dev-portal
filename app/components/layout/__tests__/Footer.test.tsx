import renderer from "react-test-renderer";
import Footer from "../Footer";

test("Footer", () => {
  const FOOTER_LINKS = {
    Dwolla: [
      { text: "About", href: "https://www.dwolla.com/about/" },
      { text: "Blog", href: "https://www.dwolla.com/updates/" },
      { text: "Pricing", href: "https://www.dwolla.com/pricing/" },
      { text: "Contact Sales", href: "https://www.dwolla.com/contact/" },
      { text: "Terms of Service", href: "https://www.dwolla.com/legal/tos/" },
      { text: "Privacy Policy", href: "https://www.dwolla.com/legal/privacy/" },
      {
        text: "Privacy Options",
        onClick: () =>
          window.Osano.cm.showDrawer("osano-cm-dom-info-dialog-open"),
      },
    ],
    Product: [
      { text: "Resources", href: "https://www.dwolla.com/resources/" },
      { text: "Features", href: "https://www.dwolla.com/features/" },
      { text: "Case Studies", href: "https://www.dwolla.com/case-studies/" },
    ],
    Developers: [
      {
        text: "API Reference",
        href: "https://developers.dwolla.com/api-reference/",
      },
      { text: "Support Forum", href: "https://discuss.dwolla.com/" },
      { text: "SDKs and Tools", href: "#sdks-and-tools" },
      { text: "Github Repos", href: "https://github.com/Dwolla/" },
    ],
  };

  const FOOTER_LEGAL_COPY = {
    title: "Financial institutions play an important role in our network.",
    description: (
      <>
        All funds transfers made using the Dwolla Platform are performed by a
        financial institution partner, and any funds held in a Dwolla Balance
        are held by a financial institution partner.{" "}
        <a
          href="https://www.dwolla.com/legal/about-our-financial-institution-partners/#legal-content"
          target="_blank"
          rel="noreferrer"
        >
          Learn more about our financial institution partners
        </a>
        .
      </>
    ),
  };

  const tree = renderer
    .create(<Footer links={FOOTER_LINKS} legal={FOOTER_LEGAL_COPY} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
