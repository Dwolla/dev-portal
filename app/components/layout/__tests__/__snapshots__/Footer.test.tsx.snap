// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer 1`] = `
<div
  className="css-rntgrl-Container eg0edpy12"
>
  <div
    className="css-549mvs-LogoContainer eg0edpy11"
  >
    <a
      href="/docs"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <img
        alt=""
        className="css-1289bd9-Logo eg0edpy10"
        src="test-image-stub"
      />
    </a>
    <div
      className="css-1neb9ej-Copyright eg0edpy9"
    >
      2024
       All Rights Reserved
    </div>
  </div>
  <div
    className="css-1g5yeab-LinkContainer eg0edpy8"
  >
    <div
      className="css-n2kuza-LinkGroup eg0edpy7"
    >
      <h3
        className="css-ve8wjm-LinkGroupHeading eg0edpy6"
      >
        Dwolla
      </h3>
      <ul
        className="css-1wxmwh7-LinkGroupList eg0edpy5"
      >
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://www.dwolla.com/about/"
          >
            About
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://www.dwolla.com/updates/"
          >
            Blog
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://www.dwolla.com/pricing/"
          >
            Pricing
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://www.dwolla.com/contact/"
          >
            Contact Sales
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://www.dwolla.com/legal/tos/"
          >
            Terms of Service
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://www.dwolla.com/legal/privacy/"
          >
            Privacy Policy
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="#"
            onClick={[Function]}
          >
            Privacy Options
          </a>
        </li>
      </ul>
    </div>
    <div
      className="css-n2kuza-LinkGroup eg0edpy7"
    >
      <h3
        className="css-ve8wjm-LinkGroupHeading eg0edpy6"
      >
        Product
      </h3>
      <ul
        className="css-1wxmwh7-LinkGroupList eg0edpy5"
      >
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://www.dwolla.com/resources/"
          >
            Resources
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://www.dwolla.com/features/"
          >
            Features
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://www.dwolla.com/case-studies/"
          >
            Case Studies
          </a>
        </li>
      </ul>
    </div>
    <div
      className="css-n2kuza-LinkGroup eg0edpy7"
    >
      <h3
        className="css-ve8wjm-LinkGroupHeading eg0edpy6"
      >
        Developers
      </h3>
      <ul
        className="css-1wxmwh7-LinkGroupList eg0edpy5"
      >
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://developers.dwolla.com/api-reference/"
          >
            API Reference
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://discuss.dwolla.com/"
          >
            Support Forum
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="#sdks-and-tools"
          >
            SDKs and Tools
          </a>
        </li>
        <li
          className="css-146nff3-LinkGroupItem eg0edpy4"
        >
          <a
            className="css-1yfy9it-LinkGroupLink eg0edpy3"
            href="https://github.com/Dwolla/"
          >
            Github Repos
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="css-1kena77-LegalContainer eg0edpy2"
  >
    <strong
      className="css-31ngrj-LegalTitle eg0edpy1"
    >
      Financial institutions play an important role in our network.
    </strong>
    <p
      className="css-b6puy4-LegalDesc eg0edpy0"
    >
      All funds transfers made using the Dwolla Platform are performed by a financial institution partner, and any funds held in a Dwolla Balance are held by a financial institution partner.
       
      <a
        href="https://www.dwolla.com/legal/about-our-financial-institution-partners/#legal-content"
        rel="noreferrer"
        target="_blank"
      >
        Learn more about our financial institution partners
      </a>
      .
    </p>
  </div>
</div>
`;
